matrix_sdk/src/sas.rs

```diff
@@ -52,7 +72,16 @@ impl Sas {
     /// Cancel the interactive verification flow.
     pub async fn cancel(&self) -> Result<()> {
         if let Some(request) = self.inner.cancel() {
-            self.http_client.send(request, self.session.clone()).await?;
+            let request: http::Request<Vec<u8>> = request.try_into()?;
+            self.http_client
```

Comment:
The abstraction layer that I have written has been lost in this PR, these methods used to use the `send()` method that just required the request and session, we might want to put the session in a the `HttpClient` wrapper as well instead of losing the abstraction layer.

This also means that the SAS verification requests aren't traced anymore.

Reply:
So I think what you're talking about is the biggest downside of this, there is no way (that I could come up with) to create an object safe trait that takes a generic with the `Endpoint` bound. Ideally, we wouldn't abstract `send_request` but `send` as it is now (`send<R: Endpoint...>`).

Reply:
I know that `async-trait` has trouble with this, it gave me some headaches as well but wouldn't a wrapper struct with the implementation work like we currently have it over [here](https://github.com/matrix-org/matrix-rust-sdk/blob/master/matrix_sdk/src/http_client.rs#L27)?

Of course instead of wrapping `reqwest::Client` we would wrap `Box<dyn HttpClientTrait>`.

Reply:
Yup, you are totally right, I'm just going to open a new PR if you don't mind. I have it so nothing changes, the Sas stuff is left nice and simple as you had it originally.